### ISSUE TITLE
`--no-warnings` in `--devel` mode doesn't quite squash the warning...

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#21652]
In developer mode (`--devel`) when compiling a program that generates warnings, I see the output:

```bash
myFile.chpl:1: In module 'myFile'
```

and then nothing else, suggesting that the logic that squashes warnings does not extend out far enough to squash that contextual information.

(that also raises the question for me:  Why do we print this module information in developer mode only?  I don't think I've missed it in user mode, but don't remember a rationale for the asymmetry).

**Comment by mppf** [comment]
@bradcray - can you show an example program with this behavior? I was recently facing orphaned "In module" / "In function" output for warnings by the dyno effort and I addressed it in #21614.

**Comment by bradcray** [comment]
Finally got back to this and merging with main, indeed found the problem resolved—thanks!  I've updated my test to lock this behavior in since it seemed to slip by us before (not surprising since testing doesn't throw `--devel`).

Michael, do you understand what the rationale is for when we do or do not print out "In module" prefixes to warnings and errors?  That is, why does a non-devel compile give:

```bash
$ chpl arrOfRanges.chpl --no-devel
arrOfRanges.chpl:3: warning: please note that this is a 1-element array of ranges; if that was your intention, add a trailing comma or recompile with '--no-warn-array-of-range' to avoid this warning; if it wasn't, you may want to use a range instead
```

while a devel compile gives:

```bash
$ chpl arrOfRanges.chpl
arrOfRanges.chpl:1: In module 'arrOfRanges':
arrOfRanges.chpl:3: warning: please note that this is a 1-element array of ranges; if that was your intention, add a trailing comma or recompile with '--no-warn-array-of-range' to avoid this warning; if it wasn't, you may want to use a range instead
```

Is the "In module" information considered overkill for the typical user, or is this a glitch that wasn't intended (and maybe not noticed because we work as developer most of the time)?
